src/sinks/util/service/concurrency.rs

```diff
@@ -1,17 +1,31 @@
+use serde::Serializer;
 use std::fmt;
 
 use serde::{
     de::{self, Unexpected, Visitor},
     Deserialize, Deserializer, Serialize,
 };
 
-#[derive(Clone, Copy, Debug, Derivative, Eq, PartialEq, Serialize)]
+#[derive(Clone, Copy, Debug, Derivative, Eq, PartialEq)]
 pub enum Concurrency {
     None,
     Adaptive,
     Fixed(usize),
 }
 
+impl Serialize for Concurrency {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match &self {
+            Concurrency::None => serializer.serialize_str("none"),
+            Concurrency::Adaptive => serializer.serialize_str("adaptive"),
+            Concurrency::Fixed(i) => serializer.serialize_u64(*i as u64),
+        }
```

Comment:
The "none" branch should never be called, since the TowerRequestConfig will skip serializing the None variant.